### PR TITLE
Add support for Firebase, basic auth actions, rough draft for core data types

### PR DESCRIPTION
**Pull request by scott-t-robinson**
- Firebase support https://console.firebase.google.com/u/0/project/shareable-mvp/overview
- General functions for authentication through google/other email
- Rough draft for core data types user, organization, groups, requests, insights
- Skeleton for adding requests/insights to Firestore database, collections schema subject to change

### CODE REVIEW

File: src/types/filedRequest.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    description?: string;
+    comments?: string[];
+    insights? : Insight[]
+}
+
+enum RequestPriority {
+    Low,
+    Medium,
+    High
+}
```
Comment (asbhaskar) [inline src/types/filedRequest.ts]: Could we replace enunms with types? e.g. ``` type RequestPriority = 'Low' | 'Medium' | 'High'  ```

File: src/types/filedRequest.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import {Insight} from "./insight";
+import {User} from "./user";
+import {Group} from "./group";
+
+export const REQUESTS_COLLECTION = "requests"
+
+export interface FiledRequest {
+    createdBy: User;
+    createDate: number;
```
Comment (asbhaskar) [inline src/types/filedRequest.ts]: Can we move interfaces into ```src/interfaces/```

File: public/index.html [hunk 3]
```diff
@@ -0,0 +1,91 @@
+<!DOCTYPE html>
```
Comment (asbhaskar) [inline public/index.html]: I don't think we need this file it looks like it's for vanillaJS apps. You also haven't included a reference for vite to use this file

File: src/actions/userAuthActions.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import { doc, setDoc } from "firebase/firestore";
+import {TEST_ORGANIZATION} from "../types/organization";
+import {TEST_GROUP} from "../types/group";
+
+const auth = getAuth(firebaseApp);
+
+export const signUp = (userCredentials: any) => {
+    createUserWithEmailAndPassword(auth, userCredentials.email, userCredentials.password)
+        .then((userCredential) => {
+           setDoc(doc(firestore, "users", userCredential.user.uid), {
```
Comment (asbhaskar) [inline src/actions/userAuthActions.ts]: Please destructure userCredentials here like ``` const {email, password, user: {uid}} = useCredentials ```

File: .firebaserc [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+  "projects": {
+    "default": "shareable-mvp"
+  }
+}
```
Comment (asbhaskar) [inline .firebaserc]: We have this information in the src/firebase.ts file, is this one necessary?

Comment (asbhaskar) [inline .firebaserc]: Do we need this? The information is also in ```src/firebase.ts```

File: firebase.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{
+  "hosting": {
+    "public": "public",
+    "ignore": [
+      "firebase.json",
+      "**/.*",
+      "**/node_modules/**"
+    ],
```
Comment (asbhaskar) [inline firebase.json]: We are not hosting the application on firebase, I don't think we need this file

File: package.json [hunk 7]
```diff
@@ -9,13 +9,16 @@
...
     "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react-dom": "^18.2.0",
+    "uuidv4": "^6.2.13"
   },
   "devDependencies": {
     "@types/react": "^18.0.26",
     "@types/react-dom": "^18.0.9",
     "@vitejs/plugin-react": "^3.0.0",
+    "firebase-tools": "^11.21.0",
```
Comment (asbhaskar) [inline package.json]: I'm not sure if firebase-tools is necessary, I don't think we need to access the CLI from the frontend

File: src/types/filedRequest.ts (hunk 1 above)
Comment (asbhaskar) [reply src/types/filedRequest.ts]: Technically I think you can do this directly in interfaces now but I think it helps with organization

**Review by asbhaskar — CHANGES_REQUESTED** [review]
Good start, just a few comments/questions below.

File: src/actions/userAuthActions.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
+import { firebaseApp, firestore } from '../firebase'
+import { getAuth, createUserWithEmailAndPassword, signInWithEmailAndPassword, GoogleAuthProvider, signInWithPopup, onAuthStateChanged } from "firebase/auth";
+import { doc, setDoc } from "firebase/firestore";
+import {TEST_ORGANIZATION} from "../types/organization";
+import {TEST_GROUP} from "../types/group";
+
+const auth = getAuth(firebaseApp);
+
+export const signUp = (userCredentials: any) => {
```
Comment (asbhaskar) [inline src/actions/userAuthActions.ts]: Please defined types for userCredentials, maybe ```{email: string, password: string, user: {id: number}}``` or something based on what you have below

**Comment by scott-t-robinson** [comment]
Pretty sure all comments are addressed. Im scared of the Vercel failure, may be because public/index.html was removed. I get a 404 when checking on Vercel, could you add me

**Comment by asbhaskar** [comment]
@scott-t-robinson I'll get back to you with the code review later, here's the vercel error:
Screen Shot 2023-02-12 at 3 42 25 PM

Unsure if we have to pay to add people to vercel but ill look that up later as well.

**Comment by scott-t-robinson** [comment]
Thanks, fixed that build error

**Review by asbhaskar — APPROVED** [review]
LGTM, I'll hook the sign in apis up and test it sometime this week, just fix the merge conflicts

**Pull request merged**